### DISCUSSION TITLE
JSON API caching

### DISCUSSION THREAD

**Discussion by sayan-discovery** [grafana/grafana-json-datasource#81] [Q&A] [answered]
Hi Team.

We are using JSON API plugin where we need to cache the values coming from a particular query. 
Issue 1>We have used a variable value in the API params. But we are observing that when the cache time is 5hrs, even if the variable value is changing, the query is not executing.. 
Issue 2>We have used Stat value as visualisation in a panel that has 3 queries. WIth cache time as 5 hours, we are seeing all queries display same value. Actually it should display 3 different values for 3 queries.

Can you provide a solution to this?

**Comment by marcusolsson** [comment, marked as answer]
Hey, and thanks for using the JSON plugin! 

- Issue 1: Currently, the cache key is the URL in the data source config and the Path from the query editor. I don't believe there should be any issues with adding the query parameters to the cache key as well. Could you submit an issue for this?
- Issue 2: I'm not sure I understand what you would like to see here. If the base URL + path is the same for all the queries, they will get the same (cached) response. Assuming they differ in query params, this should work as intended if we add query params to the cache key.

**Reply by mikeymclellan** [reply]
I'm seeing the same issue where we have the same `path` across multiple single stat panels and occasionally they all show the same numbers. I suspect it's because they're sharing the same cache key, however `params` is different on each panel. So I'm guessing `params` needs to be added to the cache key?

**Reply by mikeymclellan** [reply]
Ahh looks like that was done [3 months ago](https://github.com/marcusolsson/grafana-json-datasource/commit/7124c85e356b2084e3cf052102dc3a1f2b6fafde). I just needed to upgrade! Thanks :)

**Reply by mikeymclellan** [reply]
Sorry, I was wrong, the update to 1.2.1 didn't fix the issue. Changing the cache time to 0s seems to be the best work-around for us currently.

**Comment by SerjNikitin** [comment]
Hi, i have a similar problem, the cache is not updated when changing the variables that are passed in the request body with a caching time of 5 minutes, but if it is lowered to 10 seconds, we get the expected result, i don't understand why this is happening

**Reply by yesoreyeram** [reply]
caching is not considered with body yet. 

https://github.com/grafana/grafana-json-datasource/blob/cb9c3ce7fa7055b4429a0bef660736a0bdc6c37e/src/api.ts#L77
